src/proxy/http/orig_proto.rs

```diff
@@ -48,66 +48,58 @@ where
     }
 
     fn call(&mut self, mut req: Self::Request) -> Self::Future {
-        let mut downgrade_response = false;
-
-        if req.version() != http::Version::HTTP_2 {
-            debug!("upgrading {:?} to HTTP2 with orig-proto", req.version());
-
-            // absolute-form is far less common, origin-form is the usual,
-            // so only encode the extra information if it's different than
-            // the normal.
-            let was_absolute_form = h1::is_absolute_form(req.uri());
-            if !was_absolute_form {
-                // Since the version is going to set to HTTP_2, the NormalizeUri
-                // middleware won't normalize the URI automatically, so it
-                // needs to be done now.
-                h1::normalize_our_view_of_uri(&mut req);
-            }
-
-            let val = match (req.version(), was_absolute_form) {
-                (http::Version::HTTP_11, false) => "HTTP/1.1",
-                (http::Version::HTTP_11, true) => "HTTP/1.1; absolute-form",
-                (http::Version::HTTP_10, false) => "HTTP/1.0",
-                (http::Version::HTTP_10, true) => "HTTP/1.0; absolute-form",
-                (v, _) => unreachable!("bad orig-proto version: {:?}", v),
-            };
-            req.headers_mut().insert(
-                L5D_ORIG_PROTO,
-                HeaderValue::from_static(val)
-            );
-
-            // transfer-encoding is illegal in HTTP2
-            req.headers_mut().remove(TRANSFER_ENCODING);
-
-            *req.version_mut() = http::Version::HTTP_2;
-            downgrade_response = true;
+        if req.version() == http::Version::HTTP_2 || h1::wants_upgrade(&req) {
+            // Just passing through...
+            return self.inner.call(req).map(|res| res)
```

Comment:
heads up @seanmonstar  -- i moved the non-upgrade logic into an early return so that the control flow is a bit easier (for me) to follow